src/storage/v2/inmemory/storage.cpp

```diff
@@ -807,6 +812,25 @@ utils::BasicResult<StorageManipulationError, void> InMemoryStorage::InMemoryAcce
 
     {
       std::unique_lock<utils::SpinLock> engine_guard(storage_->engine_lock_);
+
+      if (storage_->config_.salient.items.enable_label_index_auto_creation) {
+        storage_->labels_to_auto_index_.WithLock([&](auto &label_indices) {
+          for (const auto label : label_indices) {
+            CreateIndex(label, false);
+          }
+          label_indices.clear();
+        });
+      }
+
+      if (storage_->config_.salient.items.enable_edge_type_index_auto_creation) {
+        storage_->edge_types_to_auto_index_.WithLock([&](auto &edge_type_indices) {
+          for (const auto edge_type : edge_type_indices) {
+            CreateIndex(edge_type, false);
+          }
+          edge_type_indices.clear();
+        });
+      }
+
```

Comment:
I might not understand something here so let me phrase it as question.

What happens if there is other transaction which is currently updating creating more EDGE_TYPE edges? I might be missing something but from what I get, you first collect all edge types for certain transaction in system and then you create them at transaction end, but what if other transaction is now emplacing 10000 more new edges with that edge types, I think we will miss these edge types in index. I am only 60% sure I got  everything correctly, but that is my question

Reply:
@antoniofilipovic 
Shouldn't be an issue. `labels_to_auto_index_` and `edge_types_to_auto_index_` are `memgraph::utils::synchronized` datastructures updating and reading from them are blocking concurrent threads to operate on them. In the case edge-types, once we get to line 826 we take a lock to the datastructure, and that will make the other thread(s) that are creating the edges with the same edge-type block and progress until they actually hold the lock. And once they actually acquire the lock using operator->() (since it is overloaded on `utils::synchronized)` they can continue execution without any issues.